src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -120,24 +120,8 @@ private static void EncodeToUtf16_Vector128(ReadOnlySpan<byte> bytes, Span<char>
                 uint block = Unsafe.ReadUnaligned<uint>(
                     ref Unsafe.Add(ref MemoryMarshal.GetReference(bytes), pos));
 
-                // TODO: Remove once cross-platform Shuffle is landed
-                // https://github.com/dotnet/runtime/issues/63331
-                [MethodImpl(MethodImplOptions.AggressiveInlining)]
-                static Vector128<byte> Shuffle(Vector128<byte> value, Vector128<byte> mask)
-                {
-                    if (Ssse3.IsSupported)
-                    {
-                        return Ssse3.Shuffle(value, mask);
-                    }
-                    else if (!AdvSimd.Arm64.IsSupported)
-                    {
-                        ThrowHelper.ThrowNotSupportedException();
-                    }
-                    return AdvSimd.Arm64.VectorTableLookup(value, mask);
-                }
-
                 // Calculate nibbles
-                Vector128<byte> lowNibbles = Shuffle(
+                Vector128<byte> lowNibbles = Vector128.Shuffle(
```

Comment:
We should double check the codegen here and that the JIT sees `shuffleMask` and `indices` below as `GT_CNS_VEC` during import.

Its currently an import only check and I know sometimes we won't see it as a `CNS_VEC` until closer to morph unless its directly a `Vector128.Create(....)` call.

This is codegen that I'll end up improving more in .NET 8, but its a bit close to fixup for .NET 7

Reply:
Ah, didn't know that Vector128.Shuffle currently optimized only for constant input, the 2nd call operates on top of variable indices so I assume it won't work currently